### DISCUSSION TITLE
ESP8266 compile from source for 1M flash

### DISCUSSION THREAD

**Discussion by pabou38** [micropython/micropython#13454] [ESP8266] [answered]
Hello.
I am trying to use a couple of ESP8266 with 1MB flash (better recycle than buy new stuff)

Github has indeed an official firmware for 1MB module (which also support a small file system). In this firmware, esp.flash_user_start() is about 600k (so leaving about 400k for file system)

HOWEVER, I cannot recreate such a configuration by compiling from source (using any of the manifest included as example). esp.flash_user_start()  is always 1Mb

QUESTION: is there a manifest I can use ? or (more likely) what should I modify in the source to compile a firmware for 1MB flash and a small file system

Thanks you!

**Comment by robert-hh** [comment]
Did you have the setting of BOARD_VARIANT=FLASH_1M on the make command line?

**Reply by pabou38** [reply, marked as answer]
That worked. Thanks !!. I should have RTFM more carefully, as this is mentioned in the ports/esp8266 section of github.

This indeed creates a firmware with  ~600K of firmware and the rest (400k) for filesystem.

NOTE: in my case, 600K seems not enough to freeze all of my module. Make fails with irom overflows 
(I freeze a lot of code to run it from flash (thanks's harvard), to avoid RAM shortage)

Ideally, I would like to "set the cursor" between firmware and file system, 

But this is not a critical issue. the current workaround is to put everything in flash (as frozen bytecode) with no file system at all.

thanks again
